apps/privacy-panel/js/tc/applications.js

```diff
@@ -0,0 +1,128 @@
+/**
+ * Transparency Control -- Application List panel.
+ *
+ * @module TcApplicationsPanel
+ * @return {Object}
+ */
+define([
+  'panels',
+  'app_list',
+  'tc/app_details'
+],
+
+function(panels, appList, appDetails) {
+  'use strict';
+
+  var _appListContainer;
+
+  /**
+   * TC-Applications panel
+   *
+   * @constructor
+   */
+  function TcApplicationsPanel() {}
+
+  TcApplicationsPanel.prototype = {
+
+    /**
+     * Initialize the Applications panel and its subpanel
+     *
+     * @method init
+     * @param {Object} permissionTable  List of supported permissions.
+     */
+    init: function init(permissionTable) {
+      _appListContainer = document.getElementById('tc-appList');
+      var sortKeySelect = document.getElementById('tc-sortKey');
+
+      var refreshAppList = function refreshAppList() {
+        this.renderAppList(sortKeySelect.value);
+      }.bind(this);
+      sortKeySelect.addEventListener('change', refreshAppList);
+      window.addEventListener('applicationinstall', refreshAppList);
+      window.addEventListener('applicationuninstall', refreshAppList);
+
+      appList.init(permissionTable).then(this.renderAppList.bind(this),
+          error => console.error(error));
+
+      appDetails.init();
+    },
+
+    /**
+     * Render the Applications panel.
+     *
+     * @method renderAppList
+     * @param {String} sortKey [optional]  Either 'name', 'trust', 'vendor'.
+     */
+    renderAppList: function renderAppList(sortKey) {
+      this._clear();
+      if (!sortKey || sortKey === 'name') {
+        // apps are already sorted by name, just display them
+        this._showAppList(appList.applications);
+      }
+      else {
+        var apps = appList.getSortedApps(sortKey);
+        // sorting by headers work because the sort key is either:
+        // - a "vendor" name, in which case it makes sense to sort by name
+        // - 'certified|privileged|web', which luckily matches the order we want
+        Object.keys(apps).sort().forEach(header => {
+          var l10nPrefix = (sortKey === 'trust') ? 'tc-trust-' : '';
+          this._showAppSeparator(header, l10nPrefix);
+          this._showAppList(apps[header], header);
+        });
+      }
+    },
+
+    _clear: function _clear() {
+      _appListContainer.innerHTML = '';
+    },
+
+    _showAppSeparator: function _showAppSeparator(separator, l10nPrefix) {
+      if (!separator) {
+        return;
+      }
+      var header = document.createElement('header');
+      var title = document.createElement('h2');
+      if (l10nPrefix) {
+        title.setAttribute('data-l10n-id', l10nPrefix + separator);
```

Comment:
Yes, we have all the l10nPrefix + separator pairs in the l10n file. The prefix is always `tc-trust-`, and the separator is either `certified`, `privileged` or `web`.